### PR TITLE
Move multidict iterators to C ext.

### LINKED ISSUE #241: Rewrite iterators with C
As a part of getting rid of Cython

### PR DESCRIPTION
**Pull request by iemelyanov**
Resolves #241

### CODE REVIEW

File: multidict/_multidict_iter.c [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+    0,                                              /* tp_members */
+    0,                                              /* tp_getset */
+    0,                                              /* tp_base */
+    0,                                              /* tp_dict */
+    0,                                              /* tp_descr_get */
+    0,                                              /* tp_descr_set */
+    0,                                              /* tp_dictoffset */
+    0,                                              /* tp_init */
+    0,                                              /* tp_alloc */
+    (newfunc)multidict_iter_new,                    /* tp_new */
```
Comment (asvetlov) [inline multidict/_multidict_iter.c]: Please take a look on `PyDictIterKey_Type` and family.
I think we don't need a `__new__` method but a C function to create an iterator.
Iterator instances should not be instantiated by user Python code but by views only.

File: multidict/_multidict_iter.c [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    -1,                          /* m_size */
+    NULL,                        /* m_methods */
+    NULL,                        /* m_reload */
+    NULL,                        /* m_traverse */
+    NULL,                        /* m_clear */
+    NULL                         /* m_free */
+};
+
+PyMODINIT_FUNC
+PyInit__multidict_iter(void)
```
Comment (asvetlov) [inline multidict/_multidict_iter.c]: We don't need a module for iterators. I'm ok with putting them into a separate `.c` file but no python module.
Right now `pair_list.c` intentionally is not a module too but a file with `init` function.
Later we'll convert it into `_multidict` C extention which is built from several .c files but is a single python module.

File: multidict/_multidict_iter.c (hunk 1 above)
Comment (iemelyanov) [reply multidict/_multidict_iter.c]: Ok. I added `__new__` just for compatibility with `_ItemsView` https://github.com/aio-libs/multidict/blob/master/multidict/_multidict.pyx#L497, etc.

And what we should do with `_ItemsView`, `_ValuesView`, `_KeysView`? Rewrite on C in this PR or separate?

File: multidict/_multidict_iter.c (hunk 2 above)
Comment (iemelyanov) [reply multidict/_multidict_iter.c]: Ok

**Comment by asvetlov** [comment]
Thanks. Sorry for delay.

**Pull request merged**